src/vm/mod.rs

```diff
@@ -1,3 +1,80 @@
 // SPDX-License-Identifier: Apache-2.0
 
 mod linux;
+
+use crate::vm::linux::{
+    ioctl::Cmd,
+    types::{Capabilities, CpuidConfig},
+};
+use kvm_ioctls::{Error, Kvm, VmFd};
+
+// Defined in linux/arch/x86/include/uapi/asm/kvm.h
+const KVM_X86_TDX_VM: u64 = 2;
+
+/// Handle to the KVM VM file descriptor
+pub struct TdxVm(VmFd);
+
+impl TdxVm {
+    /// Create a new TDX VM with KVM
+    pub fn create(kvm_fd: Kvm) -> Result<Self, Error> {
+        let vm_fd = kvm_fd.create_vm_with_type(KVM_X86_TDX_VM)?;
+        Ok(TdxVm(vm_fd))
+    }
+
+    /// Retrieve information about the Intel TDX module
+    pub fn get_capabilities(&self) -> Result<TdxCapabilities, kvm_ioctls::Error> {
```

Comment:
We should probably define an `Error` type to wrap the TDX ioctl error codes and better present what exactly went wrong in the ioctl, with the added knowledge of this being a TDX ioctl.